### PR TITLE
add trt api lib to paddle_fluid lib

### PR DESCRIPTION
**Pull request by NHZlX**
Add trt api lib to paddle_fluid lib and add ut to demo_ci.

### CODE REVIEW

File: paddle/fluid/inference/api/demo_ci/run.sh [hunk 1]
```diff
@@ -86,5 +94,25 @@ for WITH_STATIC_LIB in ON OFF; do
       fi
     done
   done
+  
+  # --------tensorrt mobilenet------
+  if [ $USE_TENSORRT == ON -a $TEST_GPU_CPU == ON ]; then
+    rm -rf *
```
Comment (Superjomn) [inline paddle/fluid/inference/api/demo_ci/run.sh:100]: really dangerous

change to `cd ..; rm -rf build; mkdir build`

File: paddle/fluid/inference/api/demo_ci/vis_demo.cc [hunk 2]
```diff
@@ -100,20 +101,34 @@ void CheckOutput(const std::string& referfile, const PaddleTensor& output) {
...
+    paddle::contrib::MixedRTConfig config;
+    config.param_file = FLAGS_modeldir + "/__params__";
+    config.prog_file = FLAGS_modeldir + "/__model__";
+    config.use_gpu = true;
+    config.device = 0;
+    config.max_batch_size = 1;
     config.fraction_of_gpu_memory = 0.1;  // set by yourself
+    predictor =
+        CreatePaddlePredictor<paddle::contrib::MixedRTConfig,
+                              PaddleEngineKind::kAutoMixedTensorRT>(config);
```
Comment (Superjomn) [inline paddle/fluid/inference/api/demo_ci/vis_demo.cc]: Here change to 

predictor =
        CreatePaddlePredictor<paddle::contrib::MixedRTConfig>(config);

File: paddle/fluid/inference/api/demo_ci/vis_demo.cc [hunk 3]
```diff
@@ -146,9 +161,12 @@ void Main(bool use_gpu) {
...
   google::ParseCommandLineFlags(&argc, &argv, true);
-  paddle::demo::Main(false /* use_gpu*/);
-  if (FLAGS_use_gpu) {
-    paddle::demo::Main(true /*use_gpu*/);
+  if (FLAGS_use_gpu && FLAGS_use_trt) {
+    paddle::demo::Main(true /*use_gpu*/, true);
+  } else if (FLAGS_use_gpu) {
+    paddle::demo::Main(true /*use_gpu*/, false);
+  } else {
+    paddle::demo::Main(false /* use_gpu*/, false /*use_tensorrt*/);
```
Comment (Superjomn) [inline paddle/fluid/inference/api/demo_ci/vis_demo.cc]: `/*use_gpu*/`

File: paddle/fluid/inference/api/demo_ci/CMakeLists.txt [hunk 4]
```diff
@@ -60,6 +61,13 @@ endif(NOT WIN32)
 include_directories("${PADDLE_LIB}/third_party/boost")
 include_directories("${PADDLE_LIB}/third_party/eigen3")
 
+if (NOT WIN32) 
+if (USE_TENSORRT AND WITH_GPU) 
+include_directories("${TENSORRT_INCLUDE_DIR}")
+link_directories("${TENSORRT_LIB_DIR}")
```
Comment (Superjomn) [inline paddle/fluid/inference/api/demo_ci/CMakeLists.txt]: indent here

File: paddle/fluid/inference/api/demo_ci/run.sh (hunk 1 above)
Comment (NHZlX) [reply paddle/fluid/inference/api/demo_ci/run.sh:100]: A new pr will fix this.

File: paddle/fluid/inference/api/demo_ci/vis_demo.cc (hunk 2 above)
Comment (NHZlX) [reply paddle/fluid/inference/api/demo_ci/vis_demo.cc]: Done

File: paddle/fluid/inference/api/demo_ci/vis_demo.cc (hunk 3 above)
Comment (NHZlX) [reply paddle/fluid/inference/api/demo_ci/vis_demo.cc]: Done

File: paddle/fluid/inference/api/demo_ci/CMakeLists.txt (hunk 4 above)
Comment (NHZlX) [reply paddle/fluid/inference/api/demo_ci/CMakeLists.txt]: Done

File: paddle/fluid/inference/CMakeLists.txt [hunk 5]
```diff
@@ -19,20 +19,27 @@ cc_library(paddle_fluid_origin DEPS ${fluid_modules} paddle_fluid_api)
 
 add_subdirectory(api)
 
+set(STATIC_INFERENCE_APIS paddle_fluid_api paddle_inference_api analysis_predictor)
+set(SHARED_INFERENCE_SRCS
+    io.cc ${CMAKE_CURRENT_SOURCE_DIR}/api/api.cc ${CMAKE_CURRENT_SOURCE_DIR}/api/api_impl.cc
```
Comment (luotao1) [inline paddle/fluid/inference/CMakeLists.txt:24]: `${CMAKE_CURRENT_SOURCE_DIR}/api/`用一个变量存一下，可以简短下代码

File: paddle/fluid/inference/api/demo_ci/vis_demo.cc [hunk 6]
```diff
@@ -100,20 +101,32 @@ void CheckOutput(const std::string& referfile, const PaddleTensor& output) {
...
+    }
+
+    VLOG(3) << "init predictor";
+    predictor =
+        CreatePaddlePredictor<NativeConfig, PaddleEngineKind::kNative>(config);
+  } else {
+    paddle::contrib::MixedRTConfig config;
+    config.param_file = FLAGS_modeldir + "/__params__";
+    config.prog_file = FLAGS_modeldir + "/__model__";
+    config.use_gpu = true;
```
Comment (luotao1) [inline paddle/fluid/inference/api/demo_ci/vis_demo.cc:123]: 123行 config.use_gpu=use_gpu

Comment (NHZlX) [reply paddle/fluid/inference/api/demo_ci/vis_demo.cc:123]: 这里无论什么情况，use gpu的值一定是true的

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**